### ISSUE TITLE
Specific white space in setup code causes blank screen

### ISSUE THREAD

**Issue by frontendr** [lukejacksonn/perflink#19]
Hi Luke,

As requested via Twitter, here's the issue for the error I found.

I created the following [test](
https://perf.link/#ZnVuY3Rpb24gaXNWYWxpZEJFTVBhcnQodmFsdWUpIHsKICByZXR1cm4gdmFsdWUgPT09IDAgfHwgISF2YWx1ZTsKfQpmdW5jdGlvbiBjcmVhdGVCRU1QYXJ0KHZhbHVlLCBwcmVmaXgpIHsKICByZXR1cm4gaXNWYWxpZEJFTVBhcnQodmFsdWUpID8gYCR7cHJlZml4fSR7dmFsdWV9YCA6ICIiOwp9CmZ1bmN0aW9uIEJFTShibG9jaywgZWxlbWVudCA9IG51bGwsIG1vZGlmaWVyID0gbnVsbCkgewogIHRoaXMuYiA9IGJsb2NrOwogIHRoaXMuZSA9IGVsZW1lbnQ7CiAgdGhpcy5tID0gbW9kaWZpZXI7Cn0KLy8gdG9TdHJpbmcoKSB3aGljaCBjYWNoZXMgdGhlIHJlc3VsdCBpbiB0aGUgaW5zdGFuY2UKQkVNLnByb3RvdHlwZS50b1N0cmluZyA9IGZ1bmN0aW9uIHRvU3RyaW5nKCkgewogIHJldHVybiB0aGlzLnJlbmRlcmVkIHx8ICh0aGlzLnJlbmRlcmVkID0gdGhpcy5iICsgY3JlYXRlQkVNUGFydCh0aGlzLmUsICJfXyIpICsgY3JlYXRlQkVNUGFydCh0aGlzLm0sICItLSIpKTsKfTsKLy8gdG9TdHJpbmcoKSB3aGljaCBkb2Vzbid0CkJFTS5wcm90b3R5cGUudG9TdHJpbmcyID0gZnVuY3Rpb24gdG9TdHJpbmcoKSB7CiAgcmV0dXJuIHRoaXMuYiArIGNyZWF0ZUJFTVBhcnQodGhpcy5lLCAiX18iKSArIGNyZWF0ZUJFTVBhcnQodGhpcy5tLCAiLS0iKTsKfTs=/W3siY29kZSI6ImNvbnN0IGIgPSBuZXcgQkVNKCdmb28nLCAnZWwnLCAnYmFyJyk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7IiwiZXJyb3IiOmZhbHNlLCJtZWRpYW4iOjAuMDE0OTk5OTk3ODAyMDc4NzI0fSx7ImNvZGUiOiJjb25zdCBiID0gbmV3IEJFTSgnZm9vJywgJ2VsJywgJ2JhcicpO1xuYi50b1N0cmluZzIoKTtcbmIudG9TdHJpbmcyKCk7XG5iLnRvU3RyaW5nMigpO1xuYi50b1N0cmluZzIoKTsiLCJlcnJvciI6ZmFsc2UsIm1lZGlhbiI6MC4wMDk5OTk5ODg4MzM0NDIzM31d). This link works at this time and uses the following code:

```js
function isValidBEMPart(value) {
  return value === 0 || !!value;
}
function createBEMPart(value, prefix) {
  return isValidBEMPart(value) ? `${prefix}${value}` : "";
}
function BEM(block, element = null, modifier = null) {
  this.b = block;
  this.e = element;
  this.m = modifier;
}
// toString() which caches the result in the instance
BEM.prototype.toString = function toString() {
  return this.rendered || (this.rendered = this.b + createBEMPart(this.e, "__") + createBEMPart(this.m, "--"));
};
// toString() which doesn't
BEM.prototype.toString2 = function toString() {
  return this.b + createBEMPart(this.e, "__") + createBEMPart(this.m, "--");
};
```
```js
const b = new BEM('foo', 'el', 'bar');
b.toString();
b.toString();
b.toString();
b.toString();
```
```js
const b = new BEM('foo', 'el', 'bar');
b.toString2();
b.toString2();
b.toString2();
b.toString2();
```

*But* when I change the setup code to:
```js
function isValidBEMPart(value) {
  return value === 0 || !!value;
}

function createBEMPart(value, prefix) {
  return isValidBEMPart(value) ? `${prefix}${value}` : "";
}

function BEM(block, element = null, modifier = null) {
  this.b = block;
  this.e = element;
  this.m = modifier;
}
// toString() which caches the result in the instance
BEM.prototype.toString = function toString() {
  return this.rendered || (this.rendered = this.b + createBEMPart(this.e, "__") + createBEMPart(this.m, "--"));
};
// toString() which doesn't
BEM.prototype.toString2 = function toString() {
  return this.b + createBEMPart(this.e, "__") + createBEMPart(this.m, "--");
};
```
The URL changes to [this](https://perf.link/#ZnVuY3Rpb24gaXNWYWxpZEJFTVBhcnQodmFsdWUpIHsKICByZXR1cm4gdmFsdWUgPT09IDAgfHwgISF2YWx1ZTsKfQoKZnVuY3Rpb24gY3JlYXRlQkVNUGFydCh2YWx1ZSwgcHJlZml4KSB7CiAgcmV0dXJuIGlzVmFsaWRCRU1QYXJ0KHZhbHVlKSA/IGAke3ByZWZpeH0ke3ZhbHVlfWAgOiAiIjsKfQoKZnVuY3Rpb24gQkVNKGJsb2NrLCBlbGVtZW50ID0gbnVsbCwgbW9kaWZpZXIgPSBudWxsKSB7CiAgdGhpcy5iID0gYmxvY2s7CiAgdGhpcy5lID0gZWxlbWVudDsKICB0aGlzLm0gPSBtb2RpZmllcjsKfQovLyB0b1N0cmluZygpIHdoaWNoIGNhY2hlcyB0aGUgcmVzdWx0IGluIHRoZSBpbnN0YW5jZQpCRU0ucHJvdG90eXBlLnRvU3RyaW5nID0gZnVuY3Rpb24gdG9TdHJpbmcoKSB7CiAgcmV0dXJuIHRoaXMucmVuZGVyZWQgfHwgKHRoaXMucmVuZGVyZWQgPSB0aGlzLmIgKyBjcmVhdGVCRU1QYXJ0KHRoaXMuZSwgIl9fIikgKyBjcmVhdGVCRU1QYXJ0KHRoaXMubSwgIi0tIikpOwp9OwovLyB0b1N0cmluZygpIHdoaWNoIGRvZXNuJ3QKQkVNLnByb3RvdHlwZS50b1N0cmluZzIgPSBmdW5jdGlvbiB0b1N0cmluZygpIHsKICByZXR1cm4gdGhpcy5iICsgY3JlYXRlQkVNUGFydCh0aGlzLmUsICJfXyIpICsgY3JlYXRlQkVNUGFydCh0aGlzLm0sICItLSIpOwp9Ow==/W3siY29kZSI6ImNvbnN0IGIgPSBuZXcgQkVNKCdmb28nLCAnZWwnLCAnYmFyJyk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7XG5iLnRvU3RyaW5nKCk7IiwiZXJyb3IiOmZhbHNlLCJtZWRpYW4iOjAuMDEwMDAwMDE3OTM3MjcyNzg3fSx7ImNvZGUiOiJjb25zdCBiID0gbmV3IEJFTSgnZm9vJywgJ2VsJywgJ2JhcicpO1xuYi50b1N0cmluZzIoKTtcbmIudG9TdHJpbmcyKCk7XG5iLnRvU3RyaW5nMigpO1xuYi50b1N0cmluZzIoKTsiLCJlcnJvciI6ZmFsc2UsIm1lZGlhbiI6MC4wMDUwMDAwMDg5Njg2MzYzOTM1fV0=)

But it no longer works! As you can see I've only added a blank line after the closing bracket `}` of the two functions. That's the only difference. When I remove it again the URL works just fine.

The console shows 2 exceptions:
```
react-dom.js:3231 SyntaxError: Unexpected token ` in JSON at position 1
    at JSON.parse (<anonymous>)
    at index.js:10
Bg @ react-dom.js:3231 index.js:10
Uncaught (in promise) SyntaxError: Unexpected token ` in JSON at position 1
    at JSON.parse (<anonymous>)
    at index.js:10
```

Hope you can track down the issue because this is an awesome tool!

Kind regards,
Johan (Frontendr)

**Comment by lukejacksonn** [comment]
That is super weird! Thanks for the report.. now we can try figure out why it might be happening. Let me know if you find anything. The line of code I suspect it is failing at is here https://github.com/lukejacksonn/perflink/blob/master/routes/home/index.js#L10

**Comment by tamer1an** [comment]
HI @lukejacksonn i had a similar problem 

When link is opened, JSON parse errors is poping up in the console 


![Screenshot from 2019-07-04 16-42-13](https://user-images.githubusercontent.com/956869/60674598-bddbad00-9e7a-11e9-9e1c-0df13538c4e8.png)


https://perf.link/#Y29uc3QgZGF0YSA9IFsuLi5BcnJheSgxMDAwMCkua2V5cygpXTsKY29uc3QgY2FsbGJhY2sgPSBmdW5jdGlvbigpe307/W3siY29kZSI6ImRhdGEuZm9yRWFjaCh4ID0+ICh0eXBlb2YgZnVuY3Rpb24oKXt9ID09IFwiZnVuY3Rpb25cIikgPyB0cnVlIDogbnVsbCApIiwibWVkaWFuIjoyLjAwNDk5OTk4NzU0ODIxNywiZXJyb3IiOmZhbHNlfSx7ImNvZGUiOiJkYXRhLmZvckVhY2goeCA9PiAodHlwZW9mIGZ1bmN0aW9uKCl7fSA9PT0gXCJmdW5jdGlvblwiKSA/IHRydWUgOiBudWxsICkiLCJtZWRpYW4iOjEuOTk5OTk5OTkzMTMxNDk2LCJlcnJvciI6ZmFsc2V9LHsiY29kZSI6ImRhdGEuZm9yRWFjaCh4ID0+ICh0eXBlb2YgY2FsbGJhY2sgPT0gXCJmdW5jdGlvblwiKSA/IHRydWUgOiBudWxsKSIsIm1lZGlhbiI6MS45MDAwMDAwMDI5MzM2NjYsImVycm9yIjpmYWxzZX0seyJjb2RlIjoiZGF0YS5mb3JFYWNoKHggPT4gKHR5cGVvZiBjYWxsYmFjayA9PT0gXCJmdW5jdGlvblwiKSA/IHRydWUgOiBudWxsKSIsIm1lZGlhbiI6MS45MDQ5OTk5OTczNTAzODczLCJlcnJvciI6ZmFsc2V9LHsiY29kZSI6ImRhdGEuZm9yRWFjaCh4ID0+IChjYWxsYmFjayBpbnN0YW5jZW9mIEZ1bmN0aW9uKSA/IHRydWUgOiBudWxsICkiLCJtZWRpYW4iOjMuMzA1MDAwMDA1NjM5MTU4LCJlcnJvciI6ZmFsc2V9LHsiY29kZSI6ImRhdGEuZm9yRWFjaCh4ID0+IChmdW5jdGlvbigpe30gaW5zdGFuY2VvZiBGdW5jdGlvbikgPyB0cnVlIDogbnVsbCApIiwiZXJyb3IiOmZhbHNlLCJtZWRpYW4iOjMuNDM0OTk5OTkxNDQxMTQ1NX1d

**Comment by lukejacksonn** [comment]
Hey folks, sorry it took me so long to get around to fixing this. Thanks for your reports and patience 🙇 I think this is fixed by #24.

@frontendr I remade your test suite to check (see [here](http://localhost:8080#eyJ0aXRsZSI6IkJFTSB0b1N0cmluZyBDb21wYXJpc29uICIsImJlZm9yZSI6ImZ1bmN0aW9uIGlzVmFsaWRCRU1QYXJ0KHZhbHVlKSB7XG4gIHJldHVybiB2YWx1ZSA9PT0gMCB8fCAhIXZhbHVlO1xufVxuZnVuY3Rpb24gY3JlYXRlQkVNUGFydCh2YWx1ZSwgcHJlZml4KSB7XG4gIHJldHVybiBpc1ZhbGlkQkVNUGFydCh2YWx1ZSkgPyBgJHtwcmVmaXh9JHt2YWx1ZX1gIDogXCJcIjtcbn1cbmZ1bmN0aW9uIEJFTShibG9jaywgZWxlbWVudCA9IG51bGwsIG1vZGlmaWVyID0gbnVsbCkge1xuICB0aGlzLmIgPSBibG9jaztcbiAgdGhpcy5lID0gZWxlbWVudDtcbiAgdGhpcy5tID0gbW9kaWZpZXI7XG59XG4vLyB0b1N0cmluZygpIHdoaWNoIGNhY2hlcyB0aGUgcmVzdWx0IGluIHRoZSBpbnN0YW5jZVxuQkVNLnByb3RvdHlwZS50b1N0cmluZyA9IGZ1bmN0aW9uIHRvU3RyaW5nKCkge1xuICByZXR1cm4gdGhpcy5yZW5kZXJlZCB8fCAodGhpcy5yZW5kZXJlZCA9IHRoaXMuYiArIGNyZWF0ZUJFTVBhcnQodGhpcy5lLCBcIl9fXCIpICsgY3JlYXRlQkVNUGFydCh0aGlzLm0sIFwiLS1cIikpO1xufTtcbi8vIHRvU3RyaW5nKCkgd2hpY2ggZG9lc24ndFxuQkVNLnByb3RvdHlwZS50b1N0cmluZzIgPSBmdW5jdGlvbiB0b1N0cmluZygpIHtcbiAgcmV0dXJuIHRoaXMuYiArIGNyZWF0ZUJFTVBhcnQodGhpcy5lLCBcIl9fXCIpICsgY3JlYXRlQkVNUGFydCh0aGlzLm0sIFwiLS1cIik7XG59OyIsInRlc3RzIjpbeyJuYW1lIjoiRmluZCBpdGVtIDEwMCIsImNvZGUiOiJjb25zdCBiID0gbmV3IEJFTSgnZm9vJywgJ2VsJywgJ2JhcicpO1xuYi50b1N0cmluZygpOyIsInJ1bnMiOls2MTEwMDAsNzMzMDAwLDc4MzAwMCwzNDgwMDAsNzEyMDAwLDEwNTkwMDAsNDI3MDAwLDY1MDAwMCwxMjkxMDAwLDM0MzAwMCwzNDUwMDAsODU4MDAwLDEwNzYwMDAsMTUwMDAwLDE1NzYwMDAsMTMyMDAwLDk2MjAwMCw4OTgwMDAsMTYyMDAwLDEwMDAsMjEwMDAsMTMwNjAwMCw3NjMwMDAsNzAwMCwxMzU0MDAwLDQ4MzAwMCwxMTYwMDAsMTM5MDAwLDg5MzAwMCwyNTYwMDAsMTEzMTAwMCw3ODYwMDAsMTE2NDAwMCwxNTk3MDAwLDEzNzMwMDAsMjcwMDAsMTU3MTAwMCw2MTkwMDAsODQ3MDAwLDU4NjAwMCw0NDkwMDAsMTY4ODAwMCw4MjEwMDAsNjczMDAwLDE3MDAwLDEzOTAwMCwxNjAwMDAwLDEyNjgwMDAsMTM2OTAwMCw0ODUwMDAsMTI0MDAwLDE3MDAwLDEwMDAsMzM0MDAwLDEyMDAwMDAsMTM0MjAwMCw0MDYwMDAsMTcwMDAwLDQ4MjAwMCw5NzgwMDAsMTEyOTAwMCwzODUwMDAsNzAyMDAwLDEwMDAsMjUzMDAwLDQxOTAwMCwxMDE4MDAwLDE3NDAwMCw1ODUwMDAsMTAwMCwxNzA4MDAwLDE0OTYwMDAsMTM3OTAwMCw3MjIwMDAsODg0MDAwLDExNjgwMDAsMTYwMTAwMCwxMDAwLDE0MTAwMDAsODYyMDAwLDQ2MDAwLDExNjEwMDAsNjI1MDAwLDE0MTQwMDAsMjg0MDAwLDIyMzAwMCw4MDAwLDI2NjAwMCwxNDMzMDAwLDc2MzAwMCwxMDc2MDAwLDIxMDAwMCw1NzIwMDAsMTIwMDAwLDE1NzUwMDAsMTAzMDAwLDEwNTIwMDAsOTk1MDAwLDI2OTAwMCwxMzM5MDAwXSwib3BzIjo3MTE1MTB9LHsibmFtZSI6IkZpbmQgaXRlbSAyMDAiLCJjb2RlIjoiY29uc3QgYiA9IG5ldyBCRU0oJ2ZvbycsICdlbCcsICdiYXInKTtcbmIudG9TdHJpbmcyKCk7IiwicnVucyI6WzEwMDAsMTI3NzAwMCwxNTAwMDAwLDEyODAwMDAsNjg3MDAwLDYzNjAwMCw3NDcwMDAsMTUyODAwMCwxODE1MDAwLDE1MDAwLDEwODQwMDAsODUzMDAwLDEwNjUwMDAsMTE1NzAwMCwxMDAzMDAwLDE4MjAwMCwxNDA5MDAwLDk3MzAwMCw4NjQwMDAsMTQ5MjAwMCw1MTcwMDAsMTY0NDAwMCw3ODAwMCw2NTEwMDAsMTM2NzAwMCwzNDkwMDAsMTM0MDAwLDU5NzAwMCw4NjMwMDAsMjg3MDAwLDExNzYwMDAsMTIxODAwMCw4NTUwMDAsMTQ3NTAwMCwxNDEyMDAwLDExNzEwMDAsMTAxMTAwMCw0NjUwMDAsODcyMDAwLDYzMDAwMCwxNTg3MDAwLDE2NzEwMDAsNzgyMDAwLDc0MTAwMCwxNzUxMDAwLDEwMDAsMTY5NDAwMCwxMzU4MDAwLDE1NTAwMDAsNjA4MDAwLDEwMDAsNzY5MDAwLDEwMDAsMjg4MDAwLDEyNzcwMDAsMTUxNjAwMCw4NjkwMDAsMCwxNjU3MDAwLDk3MTAwMCw4NjUwMDAsMzUwMDAwLDUxODAwMCw3MjAwMCwxNTcxMDAwLDEzOTcwMDAsMTA2NTAwMCwxMzU1MDAwLDY0MTAwMCwxMjIxMDAwLDE0NTMwMDAsMTE0NjAwMCwxNDM5MDAwLDIzMjAwMCw1MzAwMDAsMTAwMzAwMCwxMDUxMDAwLDEwMDAsODcwMDAsMTY5NDAwMCwxNzY5MDAwLDEyMzMwMDAsNTI1MDAwLDEwMDAsNzU2MDAwLDE1NTIwMDAsMTAwMCwyOTAwMDAsODgzMDAwLDc1NzAwMCwxMDAwLDEwMjgwMDAsODM1MDAwLDE0NTAwMCwxNjQyMDAwLDExOTAwMCw4NDgwMDAsNzYwMDAsNTY0MDAwLDE0OTIwMDBdLCJvcHMiOjg3NjQwMH1dLCJ1cGRhdGVkIjoiMjAyMC0wNC0wMVQxNToyNzozNy43MzdaIn0%3D)).
